packages/grafbase-sdk/src/grafbase_schema.ts

```diff
@@ -0,0 +1,187 @@
+import { Model } from './model'
+import { GRelationDef } from './relation'
+import { Enum } from './enum'
+import { FieldType, GBooleanDef, GDateDef, GNumberDef, GObjectDef, GStringDef, GScalarDef } from './field/typedefs'
+import { GListDef } from './field/list'
+import { Type } from './type'
+import { GReferenceDef } from './reference'
+import { Union } from './union'
+import { Interface } from './interface'
+import { Query, QueryInput, QueryType } from './query'
+import { EnumShape, FieldShape, RelationRef } from '.'
+
+export class GrafbaseSchema {
+  enums: Enum[]
+  types: Type[]
+  unions: Union[]
+  models: Model[]
+  interfaces: Interface[]
+  queries: Query[]
+
+  constructor() {
+    this.enums = []
+    this.types = []
+    this.unions = []
+    this.models = []
+    this.interfaces = []
+    this.queries = []
+  }
+
+  public model(name: string, fields: Record<string, FieldShape>): Model {
+    const model = Object
+      .entries(fields)
+      .reduce((model, [name, definition]) => model.field(name, definition), new Model(name))
+
+    this.models.push(model)
+
+    return model
+  }
+
+  public type(name: string, fields: Record<string, GScalarDef | GListDef>): Type {
+    const type = Object
+      .entries(fields)
+      .reduce((type, [name, definition]) => type.field(name, definition), new Type(name))
+
+    this.types.push(type)
+
+    return type
+  }
+
+  public interface(name: string, types: Record<string, GScalarDef | GListDef>): Interface {
+    const iface = Object
+      .entries(types)
+      .reduce((iface, [name, definition]) => iface.field(name, definition), new Interface(name))
+
+    this.interfaces.push(iface)
+
+    return iface
+  }
+
+  public union(name: string, types: Record<string, Type>): Union {
```

Comment:
I might be missing something, but should types just be `[Type]` rather than a record - doesn't seem like the keys are doing anything

Reply:
So the interface is per notion document:

```
g.union('UserOrAddress', { user, address })
```

Keys do nothing, but I think an array works here either if we want kind of a set(ish) behavior.